src/accounting.py

```diff
@@ -592,8 +594,9 @@ def _ledger_write(self, *args):
         t = time.time()
         elems = [str(t)] + list(map(str, args))
         salt, new_hash = generate_salt_and_hash(self.last_hash, elems, self.leading_zero_count)
-        self.ledger_file.writelines(
-            ' '.join([new_hash.hexdigest(), salt] + elems) + '\n')
+        with open(ledger_path, 'a') as f:
```

Comment:
This is clever. It costs us some IO overhead that we can likely spare anyways and in return makes the ledger a bit more stable. I like it.